internal/controller/resource_manager.go

```diff
@@ -83,8 +83,8 @@ func (r *ResourceManager) DyanmicListAndWatchResources(ctx context.Context, canc
 }
 
 // DynamicListResources lists a specifed resource dynamically and sends down the current gRPC stream.
-func (r *ResourceManager) DynamicListResources(ctx context.Context, resource string) (string, CacheManager, error) {
-	cm := CacheManager{cache: make(map[string][32]byte)}
+func (r *ResourceManager) DynamicListResources(ctx context.Context, resource string) (string, Cache, error) {
```

Comment:
The `Cache` should be passed by reference.
```suggestion
func (r *ResourceManager) DynamicListResources(ctx context.Context, resource string) (string, *Cache, error) {
```